File: src/main/java/persistence/Room.java
package persistence;

import java.util.List;

public class Room {

    private final RoomNumber roomNumber;
    private final BookingsForRoom bookings;

    public Room(RoomNumber roomNumber, List<Booking> bookings) {
        this.roomNumber = roomNumber;
        this.bookings = new BookingsForRoom(roomNumber).add(bookings);
    }

    public Room(RoomNumber roomNumber) {
        this.roomNumber = roomNumber;
        this.bookings = new BookingsForRoom(roomNumber);
    }

    public RoomNumber getRoomNumber() {
        return roomNumber;
    }

    public BookingsForRoom getBookingsForRoom() { return bookings; }
}

File: src/main/kotlin/margue/de/hotelbooking/persistence/Room.kt
package margue.de.hotelbooking.persistence

import jakarta.persistence.*
import java.util.*

@Entity
class Room(
    @Id
    val roomNumber: String,

    @OneToMany(
        mappedBy = "room",
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
    )
    var bookings: MutableList<Booking> = mutableListOf(),
)
